src/code/z_parameter.c

```diff
@@ -52,38 +293,753 @@
+                    return ITEM_NONE;
+                }
+            }
+        }
+
+    } else if ((item >= ITEM_MOON_TEAR) && (item <= ITEM_MASK_GIANT)) {
+        temp = INV_CONTENT(item);
+        INV_CONTENT(item) = item;
+        if ((item >= ITEM_MOON_TEAR) && (item <= ITEM_PENDANT_MEMORIES) && (temp != ITEM_NONE)) {
+            for (i = EQUIP_SLOT_C_LEFT; i <= EQUIP_SLOT_C_RIGHT; i++) {
+                if (temp == GET_CUR_FORM_BTN_ITEM(i)) {
+                    SET_CUR_FORM_BTN_ITEM(i, item);
+                    Interface_LoadItemIconImpl(play, i);
+                    return ITEM_NONE;
+                }
+            }
+        }
+        return ITEM_NONE;
+    }
+
+    temp = gSaveContext.save.inventory.items[slot];
+    INV_CONTENT(item) = item;
+    return temp;
+}
+
+u8 Item_CheckObtainabilityImpl(u8 item) {
+    s16 i;
+    u8 slot;
+    u8 bottleSlot;
+
+    slot = SLOT(item);
+    if (item >= ITEM_STICKS_5) {
+        slot = SLOT(sExtraItemBases[item - ITEM_STICKS_5]);
+    }
+
+    if (item == ITEM_SKULL_TOKEN) {
+        return ITEM_NONE;
+
+    } else if (item == ITEM_TINGLE_MAP) {
+        return ITEM_NONE;
+
+    } else if (item == ITEM_BOMBERS_NOTEBOOK) {
+        return ITEM_NONE;
+
+    } else if ((item >= ITEM_SWORD_KOKIRI) && (item <= ITEM_SWORD_GILDED)) {
+        return ITEM_NONE;
+
+    } else if ((item >= ITEM_SHIELD_HERO) && (item <= ITEM_SHIELD_MIRROR)) {
+        return ITEM_NONE;
+
+    } else if ((item == ITEM_KEY_BOSS) || (item == ITEM_COMPASS) || (item == ITEM_DUNGEON_MAP)) {
+        if (!CHECK_DUNGEON_ITEM(item - ITEM_KEY_BOSS, gSaveContext.mapIndex)) {
+            return ITEM_NONE;
+        }
+        return item;
+
+    } else if (item == ITEM_KEY_SMALL) {
+        return ITEM_NONE;
+
+    } else if ((item == ITEM_OCARINA) || (item == ITEM_BOMBCHU) || (item == ITEM_HOOKSHOT) || (item == ITEM_LENS) ||
+               (item == ITEM_SWORD_GREAT_FAIRY) || (item == ITEM_PICTO_BOX)) {
+        if (INV_CONTENT(item) == ITEM_NONE) {
+            return ITEM_NONE;
+        }
+        return INV_CONTENT(item);
+
+    } else if ((item >= ITEM_BOMBS_5) && (item == ITEM_BOMBS_30)) {
+        //! @bug: Should be a range check: (item <= ITEM_BOMBS_30)
+        if (CUR_UPG_VALUE(UPG_BOMB_BAG) == 0) {
+            return ITEM_NONE;
+        }
+        return 0;
+
+    } else if ((item >= ITEM_BOMBCHUS_20) && (item <= ITEM_BOMBCHUS_5)) {
+        if (CUR_UPG_VALUE(UPG_BOMB_BAG) == 0) {
+            return ITEM_NONE;
+        }
+        return 0;
+
+    } else if ((item == ITEM_QUIVER_30) || (item == ITEM_BOW)) {
+        if (CUR_UPG_VALUE(UPG_QUIVER) == 0) {
+            return ITEM_NONE;
+        }
+        return 0;
+
+    } else if ((item == ITEM_QUIVER_40) || (item == ITEM_QUIVER_50)) {
+        return ITEM_NONE;
+
+    } else if ((item == ITEM_BOMB_BAG_20) || (item == ITEM_BOMB)) {
+        if (CUR_UPG_VALUE(UPG_BOMB_BAG) == 0) {
+            return ITEM_NONE;
+        }
+        return 0;
```

Comment:
Do these 0s have a special meaning?

Reply:
It's hard to say, they're also in OoT but the return of this function only ever checks against `ITEM_NONE` from what it looks like